node/node.go

```diff
@@ -152,7 +152,7 @@ func (n *Node) Start() error {
 	}
 	// Otherwise copy and specialize the P2P configuration
 	running := new(p2p.Server)
-	*running = *n.serverConfig
+	reflect.ValueOf(running).Elem().Set(reflect.ValueOf(n.serverConfig).Elem()) // Fancy *running = *n.serverConfig to prevent mutex copy
```

Comment:
I understand the intention to fix the `go vet` issue but this is not a good fix for it.
We need to find another way. I can remove the mutex as a quick workaround.